sdk/internal/recording/recording_test.go

```diff
@@ -433,6 +433,38 @@ func TestStartStop(t *testing.T) {
 	defer jsonFile.Close()
 }
 
+func TestStartStopRoutingClient(t *testing.T) {
+	os.Setenv("AZURE_RECORD_MODE", "record")
+	defer os.Unsetenv("AZURE_RECORD_MODE")
+
+	err := Start(t, packagePath, nil)
+	require.NoError(t, err)
+
+	client, err := GetRoutingHTTPClient(t, nil)
+	require.NoError(t, err)
+
+	req, err := http.NewRequest("POST", "https://localhost:5001", nil)
+	require.NoError(t, err)
+
+	req.Header.Set(UpstreamURIHeader, "https://www.bing.com/")
+	req.Header.Set(ModeHeader, GetRecordMode())
+	req.Header.Set(IDHeader, GetRecordingId(t))
+
+	resp, err := client.Do(req)
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	require.NotNil(t, GetRecordingId(t))
+
+	err = Stop(t, nil)
+	require.NoError(t, err)
+
+	// Make sure the file is there
+	jsonFile, err := os.Open("./testdata/recordings/TestStartStopRoutingClient.json")
```

Comment:
It looks like we're checking in this file so what is this test really showing?

Should the `testdata/recordings/TestStartStopRoutingClient.json` not be checked in? Or can you check that the file has actually been updated?

Reply:
Updated to check that the URLs are changed appropriately.

Reply:
The checking part looks good.

How do you prove the file was written by your test and wasn't the older checked-in file?

Reply:
Removed the recording file from the git history. Now each test will create a new file and delete it in clean up.